integ/src/nodegroup_provider.rs

```diff
@@ -545,26 +545,25 @@ async fn confirm_nodegroup_deleted(
 fn launch_template_exists(
     result: &Result<DescribeLaunchTemplatesOutput, SdkError<DescribeLaunchTemplatesError>>,
 ) -> bool {
-    if let Err(SdkError::ServiceError { err, raw: _ }) = result {
-        if matches!(&err.kind, DescribeLaunchTemplatesErrorKind::Unhandled(_)) {
-            return false;
-        }
-    }
-    true
+    // The DescribeLaunchTemplatesErrorKind as of 0.22 only supports "Unhandled" errors.
+    // So any error should result in a failure since _all_ errors, including if the launch
+    // template doesn't exists, results in an unhandled error
+    result.is_ok()
 }
 
 fn instance_profile_exists(
     result: Result<GetInstanceProfileOutput, SdkError<GetInstanceProfileError>>,
 ) -> bool {
-    if let Err(SdkError::ServiceError { err, raw: _ }) = result {
-        if matches!(
-            &err.kind,
-            GetInstanceProfileErrorKind::NoSuchEntityException(_)
-        ) {
-            return false;
-        }
+    match result {
```

Comment:
Just a tip, no need to actually change the code to this, but matches on `Result` and `Option` types are often simplified or expressed in a more "functional" style by using some of the methods on those types, e.g.

```rust
result
    .map(|_| true)
    .unwrap_or_else(|e| !matches!(
        e.into_service_error(),
        GetInstanceProfileError {
            kind: GetInstanceProfileErrorKind::NoSuchEntityException(_),
            ..
        }
    ))


```

Reply:
That would work - I was mostly following [the advice laid out in the aws-sdk's guidance on error handling](https://github.com/awslabs/aws-sdk-rust/issues/657) and used a `match`. Thanks for the tip!!